### DISCUSSION TITLE
How to use UpdateMany with multiple values

### DISCUSSION THREAD

**Discussion by rodolphoasb** [prisma/prisma#12254] [Q&A] [answered]
Hey, how are y'all doing?!

Let's say that I have a todo list and these todos can be placed inside groups, so I would have these two models:

```
model TodoGroup {
  id          String   @id @default(cuid())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @default(now()) @updatedAt
  name        String   @unique
  description String?
  order       Int
  todos       Todo[]
}

model Todo {
  id          String    @id @default(cuid())
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @default(now()) @updatedAt
  order       Int?
  todoGroup   TodoGroup @relation(fields: [todoGroupId], references: [id])
  todoGroupId String
}
```

And each TodoGroup has a UP and DOWN arrow, to change their order on the frontend (react) and I also want to store the order inside the database to keep the UX consistent.

How would I change the order of two TodoGroups at the same time? Because if I have a "TodoGroup A" with order: 1 and I click on DOWN arrow, I want the order to become 2, and the order of "TodoGroup B" becomes 1.

**Comment by nurul3101** [comment, marked as answer]
Hey @rodolphoasb 👋 
In this particular use case, I don't think the `updateMany` method would work. `updateMany` is for updating many records with the same data. 
For your use case, you would need to use two separate `update` calls.
Something like this:
```javascript
await prisma.todoGroup.update({
      where: {
        order: 1,
      },
      data: { order: 2 },
    });

await prisma.todoGroup.update({
      where: {
        order: 2,
      },
      data: { order: 1 },
    }); 
```
[Transactions](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#the-transaction-api) can also be useful for your use case.

**Reply by rodolphoasb** [reply]
Hey, @nurul3101! Thanks for your suggestion...

Correct me if I'm wrong, but in this case, two separate updates wouldn't work because I would need to change the `order` inside `TodoGroup Model` to be `@unique` so I can use the `where` (updates require that's the value inside `where` to be `@unique`), and on the first call of update those values would match (two order:2 values) and then break...

I'll try to use transactions and I let you know the result...

**Reply by nurul3101** [reply]
Yes that's true you would need to make `order` a unique attribute to be able to use in `where` clause.
Sure, let me know if you face any issues by using transactions.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
